### PR TITLE
Clamp setEffectDensity to max allowed density for the current graphic setting

### LINKED ISSUE #785: [Request] setEffectDensity() to maximum instead of "density out of bounds" debug message
(Moved from https://bugs.mtasa.com/view.php?id=9235)
**Description**
setEffectDensity can return a debug warning about "density out of bounds" when a player has low fx settings set.

**To reproduce**
local x, y, z = getElementPosition(localPlayer)
local effect = createEffect("camflash", x, y, z, 0, 0, 0, 300)
setEffectDensity (effect, 2)

Will output a debug message if you are using low or medium fx setting.

**Expected behaviour**
Instead of a debug warning it should just set it to the max that is allowed for the current graphic setting.

**Screenshot**
http://i.imgur.com/ZmMaDdt.jpg

**Latest MTA Nightly**

**Additional context**
FX setting - maximum effect density without outputting debug warning
Low ---------- 1
Medium ----- 1.5
High --------- 2
Very High --- 2

Current workaround that only causes 1 debug message when joining:

local maxDensity = 2
function onStart()
if not setEffectDensity(createEffect("camflash", 0, 0, 0), maxDensity) then
    maxDensity = 1
end
end
addEventHandler("onClientResourceStart", resourceRoot, onStart)

### PR DESCRIPTION
**Pull request by StrixG**
Fixes #785

### CODE REVIEW

**Review by qaisjp — CHANGES_REQUESTED** [review]
I think we should keep an error if the density exceeds the global maximum (i.e, the largest possible density value across all machines). We should still keep your clamping to the "local maximum" (i.e the user's setting) though.

**Comment by StrixG** [comment]
> I think we should keep an error if the density exceeds the global maximum (i.e, the largest possible density value across all machines). We should still keep your clamping to the "local maximum" (i.e the user's setting) though.

Done.

File: Client/mods/deathmatch/logic/CClientEffect.cpp [hunk 1]
```diff
@@ -71,7 +71,7 @@ float CClientEffect::GetEffectSpeed() const
 
 bool CClientEffect::SetEffectDensity(float fDensity)
 {
-    if (fDensity >= 0)
+    if (fDensity >= 0.0f && fDensity <= 2.0f)
```
Comment (qaisjp) [inline Client/mods/deathmatch/logic/CClientEffect.cpp:74]: Is `2.0f` available as a constant anywhere? Like `MAX_EFFECT_DENSITY` or something?

Comment (StrixG) [reply Client/mods/deathmatch/logic/CClientEffect.cpp:74]: I have not found any related constants.

**Review by qaisjp — APPROVED** [review]
code LGTM

**Pull request merged**